packages/@uppy/companion/src/server/controllers/connect.js

```diff
@@ -66,3 +43,40 @@ module.exports = function connect(req, res) {
   // Now we redirect to grant's /connect endpoint, see `app.use(Grant(grantConfig))`
   res.redirect(req.companion.buildURL(`/connect/${oauthProvider}${qs}`, true))
 }
+
+
+/**
+ * initializes the oAuth flow for a provider.
```

Comment:
Needs prettier formatting

```suggestion
 * Initializes the oAuth flow for a provider.
 * After the user successfully logged-in in a different tab we sent the auth token back to Uppy with `window.postMessage()`. 
 * To prevent other tabs and unauthorized origins from accessing the token we reuse origin(s) from`corsOrigins` to limit the scope of `postMessage()`, which can can also take an origin as an argument.
 * We support multiple origins in `corsOrigins`, we have to figure out which origin is currently used.  Because the OAuth window was opened with `window.open()`, which doesn't consider the request cross origin, we don't have a `Origin` header to work with.
 * That's why the client sends the origin as a base64 encoded parameter so we can check if it matches origins from `corsOrigins` and use that as the argument to `postMessage()`. 
```